src/EditorFeatures/Core.Wpf/InlineErrors/InlineErrorAdornmentManagerProvider.cs

```diff
@@ -0,0 +1,71 @@
+using Microsoft.CodeAnalysis.Editor.InlineErrors;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Options;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Text.Classification;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.Adornments
+{
+    [Export(typeof(IWpfTextViewCreationListener))]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    [TextViewRole(PredefinedTextViewRoles.Document)]
+    internal class InlineErrorAdornmentManagerProvider : AbstractAdornmentManagerProvider<InlineErrorTag>
+    {
+        private const string LayerName = "RoslynInlineErrors";
+        private readonly IClassificationFormatMapService _classificationFormatMapService;
+        private readonly IClassificationTypeRegistryService _classificationTypeRegistryService;
+
+        [Export]
+        [Name(LayerName)]
+        [ContentType(ContentTypeNames.RoslynContentType)]
+        [Order(After = PredefinedAdornmentLayers.Selection, Before = PredefinedAdornmentLayers.Squiggle)]
+        internal readonly AdornmentLayerDefinition InlineErrorLayer;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+        public InlineErrorAdornmentManagerProvider(
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+            IThreadingContext threadingContext,
+            IViewTagAggregatorFactoryService tagAggregatorFactoryService,
+            IAsynchronousOperationListenerProvider listenerProvider,
+            IClassificationFormatMapService classificationFormatMapService,
+            IClassificationTypeRegistryService classificationTypeRegistryService
+            ) : base(threadingContext, tagAggregatorFactoryService, listenerProvider)
+        {
+            _classificationFormatMapService = classificationFormatMapService;
+            _classificationTypeRegistryService = classificationTypeRegistryService;
+        }
+
+        protected override string FeatureAttributeName => FeatureAttribute.InlineErrors;
+
+        protected override string AdornmentLayerName => LayerName;
+
+        public override void TextViewCreated(IWpfTextView textView)
+        {
+            if (textView == null)
+            {
+                throw new ArgumentNullException(nameof(textView));
+            }
+
+            if (!textView.TextBuffer.GetFeatureOnOffOption(EditorComponentOnOffOptions.Adornment))
+            {
+                return;
+            }
```

Comment:
move shared logic up.

Reply:
Right, was going to remove once adornment manager PR went in